### DISCUSSION TITLE
Passing the background color as an environment light?

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#297] [Feedback & Feature Requests]
In the default render, could you pass the result of the background color which normally gets added to the screen after the default render layer as an environment light instead so that its color can affect the objects in the scene? I think this would be nice to have for specular shaders which otherwise or without SSR can look a bit dull.

**Comment by pragma37** [comment]
Ideally, this would mean implementing a *World/Background* cube-map as a built-in concept, or supporting cube-maps as a pass input.

For now, you could render your background to an hdri-like equirectangular panorama texture and use that as a pass input.

Here's a quickly hacked example of how to use an hdri for diffuse and specular lighting:
https://github.com/bnpr/Malt/discussions/246#discussioncomment-2163943
